x-pack/filebeat/module/zoom/webhook/_meta/fields.yml

```diff
@@ -0,0 +1,647 @@
+      description: >
+        Password related to the meeting
+    - name: phone.id
+      type: keyword
+      description: >
+        Unique ID for the phone or conversation
+    - name: phone.user_id
+      type: keyword
+      description: >
+        UserID for the phone owner related to a Call Log being completed
+    - name: phone.download_url
+      type: keyword
+      description: >
+        Download URL for the voicemail
+    - name: phone.ringing_start_time
+      type: date
+      description: >
+        The timestamp when a ringtone was established to the callee
+    - name: phone.connected_start_time
+      type: date
+      description: >
+        The date and time when a ringtone was established to the callee
+    - name: phone.answer_start_time
+      type: date
+      description: >
+        The date and time when the call was answered
+    - name: phone.call_end_time
+      type: date
+      description: >
+        The date and time when the call ended
+    - name: phone.call_id
+      type: keyword
+      description: >
+        Unique ID of the related call
+    - name: phone.duration
+      type: long
+      description: >
+        Duration of a voicemail in minutes
+    - name: phone.caller_number
+      type: keyword
+      description: >
+        Caller number related to the voicemail
+    - name: phone.caller_name
+      type: keyword
+      description: >
+        Caller name related to the voicemail
+    - name: phone.caller_number_type
+      type: long
+      description: >
+        Caller type related to the voicemail, can either be 1(Internal) or 2(External)
+    - name: phone.caller_user_id
+      type: keyword
+      description: >
+        UserID of the caller related to the voicemail
+    - name: phone.callee_user_id
+      type: keyword
+      description: >
+        UserID of the callee related to the voicemail
+    - name: phone.callee_extension_type
+      type: keyword
+      description: >
+        Extension type of the calle related to the voicemail, can be user, callQueue, autoReceptionist or shareLineGroup
+    - name: phone.caller.user_id
+      type: keyword
+      description: >
+        UserID of the person which initiated the call
+    - name: phone.caller.phone_number
+      type: keyword
+      description: >
+        Phone Number of the caller related to the call
+    - name: phone.caller.extension_number
+      type: keyword
+      description: >
+        Extension number of the caller
+    - name: phone.caller.timezone
+      type: keyword
+      description: >
+        Timezone of the caller
+    - name: phone.caller.device_type
+      type: keyword
+      description: >
+        Device type used by the caller
+    - name: phone.callee_id
+      type: keyword
+      description: >
+        UserID of the related callee of the voicemail
+    - name: phone.callee.user_id
+      type: keyword
+      description: >
+        UserID of the related callee of a call
+    - name: phone.callee_name
```

Comment:
The issue is that fields nested under callee and caller is related to phone actions, while the ones not nested is related to voicemail.

Personally I agree with your opinion on this, and I can convert it to move them under the nested fields for sure :)

Reply:
Yeah even if other sources have imperfect mappings because `[reasons]`, creating a module for them is a good opportunity to fix them, at least on our side :-) Hopefully the fix for this is not too onerous